src/StatusQ/Controls/StatusTagSelector.qml

```diff
@@ -0,0 +1,121 @@
+import QtQuick 2.14
+import QtQuick.Layouts 1.12
+import QtQuick.Controls 2.14
+
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+
+//TODO check colors, sizes, margins, general clean up
+Item {
+    id: root
+
+    implicitWidth: 448
+    implicitHeight: 44
+
+    property alias textEdit: edit
+    property alias text: edit.text
+    property string warningText: ""
+    property int nameCountLimit: 5
+    property ListModel namesModel: ListModel { }
+
+    function find(model, criteria) {
+      for (var i = 0; i < model.count; ++i) if (criteria(model.get(i))) return model.get(i);
+      return null;
+    }
+
+    function insertTag(name, id) {
+        if (!find(namesModel, function(item) { return item.publicId === id }) && namesModel.count < root.nameCountLimit) {
+            namesModel.insert(namesModel.count, {"name": name, "publicId": id});
+            edit.clear();
+        }
+    }
+
+    Rectangle {
+        anchors.fill: parent
+        radius: 8
+        color: Theme.palette.baseColor2
+        border.width: 1
+        border.color: edit.activeFocus ? Theme.palette.primaryColor2 : "transparent"
+
+        RowLayout {
+            anchors.fill: parent
+            anchors.leftMargin: 16
+            anchors.rightMargin: 16
+            spacing: 8
+            StatusBaseText {
+                Layout.preferredWidth: 22
+                Layout.alignment: Qt.AlignVCenter | Qt.AlignLeft
+                color: Theme.palette.baseColor1
+                text: qsTr("To: ")
```

Comment:
This still needs to be exposed. Probably as something like `toLabel`.

Remember that Status Desktop replaces `qsTr()` with `qsTrId()` in the sources (including vendors). That's why we expose all i18n'able labels so consumers of components can decide how they want to do i18n (e.g. using `qsTr()` or any other method)

Reply:
exposed :)